fabtests/functional/unexpected_msg.c

```diff
@@ -71,9 +79,25 @@ static int alloc_bufs(void)
 	rx_buf = buf;
 	tx_buf = (char *) buf + rx_size * concurrent_msgs;
 
-	if (fi->domain_attr->mr_mode & FI_MR_LOCAL) {
-		ret = fi_mr_reg(domain, buf, buf_size, FI_SEND | FI_RECV,
-				 0, FT_MR_KEY, 0, &mr, NULL);
+	no_mr_reg = (((!(fi->domain_attr->mr_mode & FI_MR_LOCAL) &&
```

Comment:
I copied this logic from https://github.com/ofiwg/libfabric/blob/e0dc241d428dd898f0af7a49fd1287a10175f82e/fabtests/common/shared.c#L413-L418